cpp/oneapi/dal/detail/hash_map.hpp

```diff
@@ -71,6 +71,14 @@ class hash_map {
     }
 
     ~hash_map() {
+        for (std::int64_t i = 0; i < capacity_; i++) {
```

Comment:
dal::detail::threader_for or is it too much overhead ? @Alexandr-Solovev

Reply:
I'd like to say that its less safety here, and for me the current solution looks good